drivers: can: Fixed timeout values comparison

lucasdietrich opened the pull request:
Trivial fix of compilation error "invalid operands to binary "
when CONFIG_CAN_AUTO_BUS_OFF_RECOVERY = n
by using K_TIMEOUT_EQ macro.

Signed-off-by: Lucas Dietrich <ld.adecy@gmail.com>

Fixes #40290

alexanderwachter reviewed (approved):
Thanks!

Adecy commented:
I fixed the commit message, sorry for the several attempts, I wasn't using gitlint before now

alexanderwachter commented:
No worry. That's what we have the checks for.

alexanderwachter commented:
The buildkite issue seems unrelated to me.

henrikbrixandersen reviewed (approved):
We need a similar fix for can_mcux_flexcan.c. Care to include that as well?

lucasdietrich committed:
drivers: can: Fixed timeout values comparison

Trivial fix of compilation error "invalid operands to binary "
when CONFIG_CAN_AUTO_BUS_OFF_RECOVERY = n

Fixes #40290

Signed-off-by: Lucas Dietrich <ld.adecy@gmail.com>

Adecy commented:
@henrikbrixandersen I decided to gather both fixes in a single commit.

However, why did buildkite/zephyr previously fail ?
